x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/ConfigInitialNode.java

```diff
@@ -215,9 +212,8 @@ protected void execute(Terminal terminal, OptionSet options, Environment env) th
         X509Certificate httpCert = CertGenUtils.generateSignedCertificate(certificatePrincipal,
```

Comment:
Should we guard the certification generation in a try block so that we can attempt to cleanup on Exception ?

Reply:
The cleanup implies removing the cert files themselves. I'm not a fan of removing information that is useful to reproduce the issue.
This is one of the reasons that certs/keys are generated in new directories on every run: you can keep troubleshooting info around, but still permit a re-run.

Reply:
> I'm not a fan of removing information that is useful to reproduce the issue.

I don't know how much troubleshooting we expect users to do with half-done auto-configuration. Also, multiple auto-configuration dirs might add to the confusion instead of being helpful. I guess it mainly depends on the persona type of the user we expect to see this and whether or not they will attempt to troubleshoot. I don't know if we can provide something that satisfies _all_ but I think I'm of the view that we should provide a working config or nothing at all. We can revisit / re-iterate on the general approach over the next weeks/months, I can be swayed for sure. 

For this specific example though and at this point in the execution, if generation fails,  we have an empty auto-configuration directory, I can't see how this can help to troubleshoot anything, but I can see how it might confuse the user who will see it there.

Reply:
We didn't have the chance to resolve this, I;ll leave this as a TODO in the code

Reply:
We chatted and we agreed that we want to remove the auto-config dir in this case too